From b5a870851abc417b5aaac295c9ee2f62d14ccc54 Mon Sep 17 00:00:00 2001
Message-Id: <b5a870851abc417b5aaac295c9ee2f62d14ccc54.1587150132.git.zanussi@kernel.org>
In-Reply-To: <05c07aad7517537dec221029782bf891341897c5.1587150132.git.zanussi@kernel.org>
References: <05c07aad7517537dec221029782bf891341897c5.1587150132.git.zanussi@kernel.org>
From: Anna-Maria Gleixner <anna-maria@linutronix.de>
Date: Thu, 5 Jul 2018 12:43:18 +0200
Subject: [PATCH 418/506] Revert "timer: delay waking softirqs from the jiffy
 tick"

[ Upstream commit b5b16907c58280e015d5673dca4c6bd3fde0c348 ]

This patch was required as long as RT tasks where accounted to CFS
load but this was only a work around. Upstream Commit 17bdcf949d03
("sched: Drop all load weight manipulation for RT tasks") fixed the
accounting of RT tasks into CFS load.

Remove the patch and fix dependencies.

Signed-off-by: Anna-Maria Gleixner <anna-maria@linutronix.de>
Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Signed-off-by: Steven Rostedt (VMware) <rostedt@goodmis.org>
---
 kernel/time/timer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/time/timer.c b/kernel/time/timer.c
index 35628d539d2c..0e0bad34b018 100644
--- a/kernel/time/timer.c
+++ b/kernel/time/timer.c
@@ -1637,13 +1637,13 @@ void update_process_times(int user_tick)
 
 	/* Note: this timer irq context must be accounted for as well. */
 	account_process_tick(p, user_tick);
-	scheduler_tick();
 	run_local_timers();
 	rcu_check_callbacks(user_tick);
 #if defined(CONFIG_IRQ_WORK)
 	if (in_irq())
 		irq_work_tick();
 #endif
+	scheduler_tick();
 	if (IS_ENABLED(CONFIG_POSIX_TIMERS))
 		run_posix_cpu_timers(p);
 }
-- 
2.17.1

